bodhi/tests/server/consumers/test_masher.py

```diff
@@ -2232,11 +2232,123 @@ def test_expire_buildroot_overrides_exception(self, expire, exception_log, publi
+        self._new_mash_stage_dir = tempfile.mkdtemp()
+
+        # Since the ComposerThread is a subclass of Thread and since it is already constructed
+        # before we have a chance to alter it, we need to change its superclass to be
+        # dummy_threading.Thread so that the test suite doesn't launch real Threads. Threads cannot
+        # use the same database sessions, and that means that changes that threads make will not
+        # appear in other thread's sessions, which cause a lot of problems in these tests.
+        # Mock was not able to make this change since the __bases__ attribute cannot be removed, but
+        # we don't really need this to be cleaned up since we don't want any tests launching theads
+        # anyway.
+        ComposerThread.__bases__ = (dummy_threading.Thread,)
+        test_config = base.original_config.copy()
+        test_config['mash_stage_dir'] = self._new_mash_stage_dir
+        test_config['mash_dir'] = os.path.join(self._new_mash_stage_dir, 'mash')
+
+        mock_config = mock.patch.dict(
+            'bodhi.server.consumers.masher.config', test_config)
+        mock_config.start()
+        self.addCleanup(mock_config.stop)
+
+        os.makedirs(os.path.join(self._new_mash_stage_dir, 'mash'))
+
+        self.koji = buildsys.get_session()
+        self.koji.clear()  # clear out our dev introspection
+
+        self.tempdir = tempfile.mkdtemp('bodhi')
+        self.db_factory = base.TransactionalSessionMaker(self.Session)
+        self.masher = Masher(FakeHub(), db_factory=self.db_factory, mash_dir=self.tempdir)
+
+        # Reset "cached" objects before each test.
+        Release._all_releases = None
+        Release._tag_cache = None
+
+        self.expected_sems = 0
+        self.semmock = mock.MagicMock()
+        self.masher.max_mashes_sem = self.semmock
+
+    def tearDown(self):
+        super(TestMasher, self).tearDown()
+
+        self.assert_sems(self.expected_sems)
+
+        shutil.rmtree(self.tempdir)
+        shutil.rmtree(self._new_mash_stage_dir)
+
+    @mock.patch('bodhi.server.consumers.masher.bugs.set_bugtracker')
+    def test___init___sets_bugtracker(self, set_bugtracker):
+        """
+        Assert that Masher.__init__() calls bodhi.server.bugs.set_bugtracker().
+        """
+        Masher(FakeHub(), db_factory=self.db_factory, mash_dir=self.tempdir)
+
+        set_bugtracker.assert_called_once_with()
+
+    @mock.patch.dict('bodhi.server.config.config', {'pungi.cmd': '/does/not/exist'})
+    def test___init___missing_paths(self):
+        """__init__() should raise a ValueError if configured paths do not exist."""
+        for s in ('pungi.cmd', 'mash_dir', 'mash_stage_dir'):
+            with mock.patch.dict('bodhi.server.config.config', {s: '/does/not/exist'}):
+                with self.assertRaises(ValueError) as exc:
+                    Masher(FakeHub(), db_factory=self.db_factory, mash_dir=self.tempdir)
+
+                    self.assertEqual(
```

Comment:
Lets move this assertion outside "with" statement. Here, it will be skipped by test.